stackslib/src/chainstate/nakamoto/tests/node.rs

```diff
@@ -641,63 +763,130 @@ impl TestStacksNode {
+                        Err(e) => {
+                            error!(
+                                "Failed to process nakamoto block: {:?}\n{:?}",
+                                &e, &nakamoto_block
+                            );
+                            panic!();
+                        }
+                    }
+                } else {
+                    false
+                };
+                if accepted {
+                    test_debug!("Accepted Nakamoto block {}", &block_to_store.block_id());
+                    coord.handle_new_nakamoto_stacks_block().unwrap();
+                    processed_blocks.push(block_to_store.clone());
+
+                    if block_to_store.block_id() == block_id {
+                        // confirm that the chain tip advanced
+                        let stacks_chain_tip = NakamotoChainState::get_canonical_block_header(
+                            chainstate.db(),
+                            &sortdb,
+                        )
                         .unwrap()
                         .unwrap();
-                let nakamoto_chain_tip = stacks_chain_tip
-                    .anchored_header
-                    .as_stacks_nakamoto()
-                    .expect("FATAL: chain tip is not a Nakamoto block");
-                assert_eq!(nakamoto_chain_tip, &nakamoto_block.header);
-            } else {
-                test_debug!("Did NOT accept Nakamoto block {}", &block_id);
+                        let nakamoto_chain_tip = stacks_chain_tip
+                            .anchored_header
+                            .as_stacks_nakamoto()
+                            .expect("FATAL: chain tip is not a Nakamoto block");
+                        assert_eq!(nakamoto_chain_tip, &nakamoto_block.header);
+                    }
+                } else {
+                    if try_to_process {
+                        test_debug!(
+                            "Did NOT accept Nakamoto block {}",
+                            &block_to_store.block_id()
+                        );
+                        break;
+                    } else {
+                        test_debug!(
+                            "Test will NOT process Nakamoto block {}",
+                            &block_to_store.block_id()
+                        );
+                    }
+                }
+
+                let num_sigs = block_to_store.header.signer_signature.len();
+
+                // force this block to have a different sighash, in addition to different
+                // signatures, so that both blocks are valid at a consensus level
+                block_to_store.header.version += 1;
+                block_to_store.header.signer_signature.clear();
+
+                miner.sign_nakamoto_block(&mut block_to_store);
+                signers.sign_block_with_reward_set(&mut block_to_store, &reward_set);
+
+                while block_to_store.header.signer_signature.len() >= num_sigs {
```

Comment:
Not sure I understand here. Why do we want less signatures on this different block than before?

Reply:
We're generating semantically-equivalent but malleablized blocks here.  It does so by changing the set of signatures on the block, and changing the block's version byte (which isn't used for anything consensus-critical).  This way, these blocks will each have unique and distinct block IDs but will be in conflict with one another (i.e. necessarily belonging to different forks).